client/src/modules/email-report/modal/categories-people-list-ctrl.js

```diff
@@ -0,0 +1,126 @@
+angular.module('bhima.controllers')
+  .controller('CategoriesPeopleListController', CategoriesPeopleListController);
+
+CategoriesPeopleListController.$inject = [
+  '$state', 'EmailReportService', 'CategoriesPeopleService', 'SessionService', 'util',
+  'NotifyService', 'ScrollService', 'bhConstants', 'uiGridConstants',
+];
+
+
+/**
+ * Report group Controller
+ *
+ * @description
+ *CategoriesPeopleListController give the possibility to select a group of people
+ * from de database, they can be users, patients, donators..
+ * Once this list is displayed in a table , the user can select one of theme
+ * for eache person the application can get automaticly his email and name 
+ * 
+ * This selection is usefull if the use don't know a person email or name but he is registered
+ * in the database
+ */
+
+function CategoriesPeopleListController($state, EmailReportSvc, CategoriesPeople, Session, util, Notify, ScrollTo, bhConstants, uiGridConstants) {
+
+  var vm = this;
+  //emailReport is the object that contains : name,email, frequence and report-group
+  vm.listCategoriesPeople = CategoriesPeople.list;
+  vm.selectedCategory = {};
+
+  vm.loading = false;
+  vm.hasError = false;
+
+  function load(filters) {
+    /*
+     category people index is save in the localStorage as expained after
+     if there's no index, we can use it, in the most case when the user refresh the page
+     */
+    if (!$state.params.index_current_category) {
+      $state.params.index_current_category = localStorage.getItem('index_current_category');
+    }
+
+    //if a categry of peoplo has been selected from the list, 
+    //we can use that index in order to create an sql request whitch return a list of people
+    if ($state.params.index_current_category) {
+
+      var index = $state.params.index_current_category;
+
+      //saving the index in the localstorage because the user can refresh the page
+      // and state can be deleted
+      localStorage.setItem('index_current_category', index);
+
+      var currentCategory = CategoriesPeople.list[index];
+      vm.selectedCategory=currentCategory;
+      /*
+       we know now in what table in the database we request
+       */
+      var sql = "SELECT " + currentCategory.Columns[0] + " , " + currentCategory.Columns[1] + " FROM " + currentCategory.table;
```

Comment:
It is really dangerous to put SQL on the client, since an attacker can modify the string and do anything they want.  Please move this code to the server.